up-l2/dispatchers/README.adoc

```diff
@@ -30,95 +30,88 @@ SPDX-License-Identifier: Apache-2.0
+When a dispatcher is unable to dispatch an event for a given reason (e.g. queue full, etc...), the dispatcher is responsible to generate an RPC Response message and send it to the originator of the request. The status code is populated in the  link:../messages/v1/README.adoc#_response_message[`commstatus`] attribute of the response message using the error codes defined in https://github.com/eclipse-uprotocol/uprotocol-core-api/blob/main/src/main/proto/ustatus.proto[UCode].
 
-.commstatus Codes
-[#commstatus-codes]
-[width="100%",cols="30%,60%",options="header",]
-!===
-|*google.rpc.Code* |*Reason*
 
-|`*UNAVAILABLE*`
-|The req.v1 has expired due to the downstream uE was unavailable (ex. uDevice was disconnected). uE that issued the req.v1 MAY retry with back-off
 
-|`*DEADLINE_EXCEEDED*`
-|CE has timed out per the ttl attribute specifications defined in req.v1 event
+<<rpc-error-flow>> figure below illustrates the sequence of messages for RPC flows and the role dispatchers play in error handling.
 
-|`*PERMISSION_DENIED*`
-|source is not permitted to access sink
+.RPC Error Flow
+[#rpc-error-flow]
+image::rpc_flow.png[RPC Error Handling]
 
-|`*UNAUTHENTICATED*`
-|source does not have valid authentication credentials (ex. uE's identity does not match the source attribute)
 
-|`*RESOURCE_EXHAUSTED*`
-|The dispatcher ran out of resources (buffer full)
+=== uStreamer Throttling of Published Messages
 
-|`*INVALID_ARGUMENT*`
-|Invalid CE header attributes not covered above (ex. any mal-formatted attributes)
+Throttling is the act of reducing the publication frequency of messages published between devices. A subscriber might want to throttle publishations to reduce network bandwidth costs (e.g. cloud app subscribing to vehicle topics like GPS location). 
 
-|`*UNKNOWN*`
-|An unknown (but not critical) error has occurred
+A subscriber uses  `max_frequency` attribute of https://github.com/eclipse-uprotocol/uprotocol-core-api/blob/main/src/main/proto/core/usubscription/v3/usubscription.proto#L121[SubscribeAttributes] message to set the desired maximum publication frequency (value in Hertz), this attribute is only considered for remote topics. 
 
-|`*INTERNAL*`
-|There is a serious error has occurred not described by error codes mentioned above
-!===
 
-<<rpc-error-flow>> figure below illustrates the sequence of messages for RPC flows and the role dispatchers play in error handling.
+* *MUST* only forward published messages at the `max_frequency`
 
-.RPC Error Flow
-[#rpc-error-flow]
-image::rpc_flow.png[RPC Error Handling]
\ No newline at end of file
+If messages are received at a higher frequency than `max_frequency`:
+
+* *MUST* be dropped by the uStreamer
+* Dropped messages *MUST NOT* be added to the DLT
+
+if `max_frequency` is not set or the frequency is less than the publisher's frequency:
```

Comment:
Agree on if max_frequency is not set.  

Should it state, if max_frequency is higher than the publisher's frequency than must not throttle messages?

Assuming that is correct, my concern is that we are adding complexity into streamer logic when it is not needed in this scenario.